CMakeLists.txt

```
@@ -52,6 +52,13 @@ endif()
 # It is currently needed default while the specification is being formalized,
 # and to study the performance impact.
 option (ENABLE_OPENCL_LAYERS "Enable OpenCL Layers" ON)
+option (ENABLE_OPENCL_LAYERINFO "Enable building cllayerinfo tool" ON)
+
+if (ENABLE_OPENCL_LAYERS AND ENABLE_OPENCL_LAYERINFO)
+  set (ENABLE_OPENCL_LAYERINFO ON)
+else()
+  set (ENABLE_OPENCL_LAYERINFO OFF)
+endif()
```

Using [`CMakeDependentOption`](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html) is a bit less noisy. (Also available in CMake 3.1).
```suggestion
include(CMakeDependentOption)
cmake_dependent_option(ENABLE_OPENCL_LAYERINFO "Enable building cllayerinfo tool" ON ENABLE_OPENCL_LAYERS OFF)
```

---

Thanks @MathiasMagnus looks good to me, commiting.